osidb/serializer.py

```diff
@@ -1488,6 +1489,34 @@ class Meta:
         model = FlawCVSS
         fields = ["flaw"] + AbstractCVSSSerializer.Meta.fields
 
+    def create(self, validated_data):
+        cvss = FlawCVSS.objects.create_cvss(**validated_data)
+        flaw = validated_data["flaw"]
+
+        # Check that the CVSS score is 0.0 for flaws with no impact
+        if flaw.impact == Impact.NOVALUE and cvss.cvss_object.base_score != Decimal(
+            "0.0"
+        ):
+            raise serializers.ValidationError(
+                {"cvss_score": ["CVSS score must be 0.0 for flaws with no impact"]}
+            )
+
+        return super().create(validated_data)
+
+    def update(self, new_cvss, validated_data):
+        cvss = FlawCVSS.objects.create_cvss(**validated_data)
+        flaw = validated_data["flaw"]
+
+        # Check that the CVSS score is 0.0 for flaws with no impact
+        if flaw.impact == Impact.NOVALUE and cvss.cvss_object.base_score != Decimal(
+            "0.0"
+        ):
+            raise serializers.ValidationError(
+                {"cvss_score": ["CVSS score must be 0.0 for flaws with no impact"]}
+            )
+
+        return super().update(new_cvss, validated_data)
```

Comment:
Why is this done on the serializer level? Don't we have a validation mixin for such things? :thinking:

Reply:
Ah, I see, that is because the CVSS score is populated based on the vector by signal after the save.
https://github.com/RedHatProductSecurity/osidb/blob/master/osidb/signals.py#L89

When we implemented that, there was a kinda chicken/egg problem as to populate the score value, you need to first validate the string but the validations are run after the values were changed and altering the score in the validation would feel wrong.

Actually I am just thinking about it and it might mean that some other validations are wrong because of this functionality.

Reply:
I was able to access the `cvss_object.base_score` field on the Model (should have tried after I found the field while tinkering with the serializer), so I've moved the validation from the serializer to its own `_validate_` method